src/backend/distributed/commands/function.c

```diff
@@ -91,6 +131,277 @@ CreateFunctionDDLCommandsIdempotent(const ObjectAddress *functionAddress)
 }
 
 
+/*
+ * GetDistributionArgIndex calculates the distribution argument with the given
+ * parameters. The function errors out if no valid argument is found.
+ */
+static int
+GetDistributionArgIndex(Oid functionOid, text *distributionArgumentNameText,
+						Oid *distributionArgumentOid)
+{
+	char *distributionArgumentName = text_to_cstring(distributionArgumentNameText);
+	int distributionArgumentIndex = -1;
+
+	int numberOfArgs = 0;
+	int argIndex = 0;
+	Oid *argTypes = NULL;
+	char **argNames = NULL;
+	char *argModes = NULL;
+
+	HeapTuple proctup = NULL;
+
+	*distributionArgumentOid = InvalidOid;
+
+	proctup = SearchSysCache1(PROCOID, ObjectIdGetDatum(functionOid));
+	if (!HeapTupleIsValid(proctup))
+	{
+		elog(ERROR, "cache lookup failed for function %u", functionOid);
+	}
+
+	numberOfArgs = get_func_arg_info(proctup, &argTypes, &argNames, &argModes);
+
+	distributionArgumentName = text_to_cstring(distributionArgumentNameText);
```

Comment:
are we doing this twice? usually I think it's a bit nicer to use char * in internal functions and convert the text early on in the UDF.